addOns/custompayloads/src/main/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsMultipleOptionsTablePanel.java

```diff
@@ -103,6 +120,58 @@ public void actionPerformed(ActionEvent evt) {
         addButton(resetButton);
     }
 
+    private void addPayloadFileButton() {
+        fileButton = new JButton(ADD_MULTIPLE_PAYLOADS_BUTTON);
+        fileButton.addActionListener(
+                e -> {
+                    CustomPayload multiplePayloads = new CustomPayload(-1, true, "", "");
+                    CustomMultiplePayloadDialog dialog =
+                            new CustomMultiplePayloadDialog(
+                                    View.getSingleton().getOptionsDialog(null), multiplePayloads);
+                    dialog.pack();
+                    dialog.setVisible(true);
+                    File file = null;
+                    boolean preventDuplicates = false;
+                    if (dialog.isSaved()) {
+                        file = dialog.getFile();
+                        preventDuplicates = dialog.isPreventDuplicates();
+                    }
+                    if (file == null) {
+                        return;
+                    }
+                    try (BufferedReader txtReader = Files.newBufferedReader(file.toPath())) {
+                        String line;
+                        ArrayList<CustomPayload> payloads = new ArrayList<>();
+                        while ((line = txtReader.readLine()) != null) {
+                            CustomPayload newPayload =
+                                    new CustomPayload(multiplePayloads.getCategory(), "");
+                            newPayload.setPayload(line);
+                            if (preventDuplicates) {
+                                if (tableModel.isNotDuplicate(newPayload)) {
+                                    payloads.add(newPayload);
+                                }
+                            } else {
+                                payloads.add(newPayload);
+                            }
+                        }
+                        for (CustomPayload payload : payloads) {
+                            tableModel.setNextIdToPayload(payload);
+                            tableModel.addToTable(payload);
+                        }
+
+                    } catch (IOException er) {
+                        LOG.warn(er.getMessage(), er);
+                        JOptionPane.showMessageDialog(
+                                this,
+                                er,
```

Comment:
This should have a message like `An error occurred while importing the payloads:\n{0}`.